src/main/java/com/expediagroup/hiveberg/IcebergInputFormat.java

```diff
@@ -252,4 +263,19 @@ public CombinedScanTask getTask() {
     }
   }
 
+  private Schema schemaWithVirtualColumn(Schema schema) {
+    List<Types.NestedField> columns = new ArrayList<>(schema.columns());
+    columns.add(Types.NestedField.optional(9999, "snapshot_id", Types.LongType.get()));
```

Comment:
Does this fetch the "most recent" snapshot? Is 9999 just a high number you chose or does it have some significance?

Reply:
It'll be whichever snapshot the table is currently using/ pointing at. So usually that'll be the most recent but when time travelling that'll be an older snapshot id

Reply:
Oh and yes, 9999 is just a sufficiently large number as all schema field ID's need to be unique and I'm hoping there isn't a schema out there with 9999+ fields :')

Reply:
Yeah that would probably be much safer, I'll make that change